lib/propolis/src/enlightenment/hyperv/overlay.rs

```diff
@@ -407,26 +492,52 @@ impl ManagerInner {
         // position.
         let mut to_mapping = MappedPfn::new(to_pfn, &memctx)?;
         if let Some(to_set) = self.overlays.get(&to_pfn) {
-            if to_set.contains_kind(kind) {
+            if to_set.overlays.contains(&kind) {
                 return Err(OverlayError::KindInUse(to_pfn.into(), kind));
             }
         }
 
         // Removing the old overlay might fail if it doesn't exist in
         // `from_pfn`, but if it succeeds it will always produce the previous
         // page contents.
-        let contents = self
-            .remove_overlay(from_pfn, kind, acc_mem, RemoveReturnContents::Yes)?
-            .expect("asked for old contents");
+        self.remove_overlay(from_pfn, kind, acc_mem)?;
 
         // Adding a new overlay with a valid mapping can only fail if the PFN
         // already has an overlay of the selected kind, and that was checked
         // above.
-        self.add_overlay_using_mapping(&mut to_mapping, kind, contents)
+        self.add_overlay_using_mapping(&mut to_mapping, kind)
             .expect("already checked new PFN for an overlay of this kind");
 
         Ok(())
     }
+
+    /// Visits each of the overlaid pages known to this manager and overwrites
+    /// their original contents with the data in the supplied `contents` map.
+    fn restore_original_contents(
```

Comment:
I find the naming here a little bit confusing: when removing an overlay page, we call a function called `restore_overlaid_data` that actually puts back the previous contents of the guest page when an overlay was applied. This function, `restore_original_contents`, seems like it's called during an import (?), and sets the original contents for each page in the overlay manager to the values in the provided map --- it doesn't actually put back the original contents in the guest page, it just sets what the original contents that we _would_ restore when removing an overlay page would be. This surprised me. 

I might suggest not calling both of these operations "restore" --- this one seems more like `set_original_contents` or even just `import_original_contents` or something...

Reply:
Changed the verbs to "export" and "import" in 9ecba4e (in part to reflect that these are really only expected to be used during a migration).